### PR TITLE
Speed up binary delta patching by updating progress bar less often

### PR DESCRIPTION
**Pull request by Hyarion-SS**
* BPS and IPS progress bars are only updated every 256 iterations.

* Also adds tested but currently undocumented file mode to BPM: use GNU Coreutil's "split --bytes=2M" on source and target files, then create a BPM from the created source and target folders. This BPM will work on the source file to create the target file.

### CODE REVIEW

**Comment by SirNapkin1334** [comment]
I personally wouldn't implement this, as GodMode9 has great progress bars and these bars would be a discrepancy with the rest of GodMode9. Though it's d0k3's choice.

**Comment by Hyarion-SS** [comment]
BPS has many, many quick iterations. Our formerly 20 minute BPM was reduced to a minute or less, and the visual difference is unnoticeable.

File: arm9/source/crypto/crc32.c [hunk 1]
```diff
@@ -66,7 +66,6 @@ uint32_t crc32_calculate_from_file(FIL inputFile, unsigned int length) {
     uint32_t crc32 = ~0;
     uint8_t data;
     unsigned int br;
-    fvx_lseek(&inputFile, 0);
     for(unsigned i = 0; i < length; i++) {
         fvx_read(&inputFile, &data, 1, &br);
```
[the remark this answers is not in the document]
Comment (d0k3) [reply arm9/source/crypto/crc32.c]: Agreed with @Wolfvak - that's some easy way to speed things up. The 3DS hardware is really not made to read a large number of single bytes.

[the remark this answers is not in the document]
Comment (Hyarion-SS) [reply arm9/source/crypto/crc32.c]: Agreed, and the problem is larger than just this. I've been putting it off as this literal interpretation of memory reads to SD reads is mostly contained to fallback branches, but that makes it terribly slow for large files - where it is most needed in the first place. This particular one is an easy fix, but BPS in particular is going to require getting rid of some of the assumptions of the code that this was ported from.

In hindsight perhaps I pushed this a little early - the speedup from fewer progress bar updates is truly drastic, but if I can get buffered reads and writes working on the trickiest cases, I may be able to remove the bandaid of bpmFileMode... essentially it's offloading the work of making sure that large files will play nicely with buffering under these assumptions to the patching process.

You can merge the progress bar counters locally if you like, as it's only a few lines, but give the rest a little more time in the oven, I suppose.

[the remark this answers is not in the document]
Comment (d0k3) [reply arm9/source/crypto/crc32.c]: It does look like more than just a few lines of changes. I'm okay with merging it, tho. So, is this ready?

[the remark this answers is not in the document]
Comment (Hyarion-SS) [reply arm9/source/crypto/crc32.c]: Mm, give this one more day. I can knock out the fallback rewrite during my morning coffee (the groundwork is already written) and run tests over the day to make sure it works and see if bpmFileMode can be removed.

**Comment by Hyarion-SS** [comment]
This is undergoing quite the rewrite. I'll be closing this PR as the end product will look quite different.

**Pull request closed without merging**